components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt

```diff
@@ -96,13 +97,33 @@ class CrashReporter(
         }
     }
 
+    /**
+     * Submit a crash report to all registered services.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun submitTelemetryCrashReport(crash: Crash, then: () -> Unit = {}): Job {
+        return scope.launch {
+            telemetryServices.forEach { telemetryService ->
+                when (crash) {
+                    is Crash.NativeCodeCrash -> telemetryService.report(crash)
+                    is Crash.UncaughtExceptionCrash -> telemetryService.report(crash)
+                }
+            }
+
+            logger.info("Crash report submitted to ${services.size} telemetry services")
+            withContext(Dispatchers.Main) {
+                then()
+            }
```

Comment:
Is this callback mechanism with `then` even needed here?

Reply:
I left it in so its available if needed.  But you're right that if we can always add it in later.    I'll remove it.

Reply:
FYI this is now needed since its now called from a service.